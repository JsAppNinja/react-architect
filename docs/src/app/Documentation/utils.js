export const splitOnCaps = s => s.split(/(?=[A-Z1-9])/);

function getString(s) {
  return !(s instanceof Array) ? splitOnCaps(s) : s;
}

export const toDashedName = (s) => getString(s).map(s => s.toLowerCase()).join('-');
export const toTitle = (s) => getString(s).join(' ');
