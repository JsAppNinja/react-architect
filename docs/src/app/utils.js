import React from 'react';
export const githubHref = 'https://github.com/mlaursen/react-md';

const lorems = [
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at placerat nulla. Morbi eu erat diam. Donec malesuada mauris in neque efficitur luctus. Vestibulum porttitor elit velit, pellentesque imperdiet tellus maximus in. Phasellus consectetur lacus nunc, quis efficitur felis feugiat ac. Vivamus non erat laoreet, volutpat tellus sit amet, porttitor orci. Sed scelerisque id arcu at ornare. Suspendisse potenti. Phasellus eu posuere mi. Fusce venenatis odio dui, ac accumsan lacus dapibus et. Praesent euismod nulla at porta elementum. Nunc ullamcorper velit aliquam velit faucibus tempus.',
  'Duis ligula elit, elementum a nisl a, venenatis fermentum libero. Cras bibendum dignissim nisl, sed sodales tellus scelerisque sed. Sed at enim semper, imperdiet elit et, venenatis magna. Morbi vestibulum purus in ante placerat viverra. Nullam molestie arcu ut dolor rutrum varius. Sed vehicula laoreet volutpat. Quisque feugiat, nisl in sollicitudin consectetur, mi metus aliquet urna, ac sagittis nulla nulla in diam. Fusce id enim auctor ante fringilla auctor in et quam.',
  'Donec pulvinar ligula augue, et tincidunt velit bibendum non. Aenean id metus nunc. Donec non feugiat nisl. In accumsan ligula purus, ut vulputate elit tempus sit amet. Nullam volutpat semper lectus. Nunc a libero tellus. Duis posuere nisi diam, nec euismod risus interdum sit amet. Pellentesque at lacus bibendum, cursus ante non, maximus arcu. Morbi id leo nulla. Integer quis feugiat quam. Etiam auctor pharetra vestibulum.',
  'Nulla sodales metus metus. Sed a metus quis ipsum sagittis placerat. Nunc sit amet molestie purus, nec rutrum quam. Praesent elementum risus eget mattis condimentum. Morbi dignissim, dolor et finibus sollicitudin, nulla enim molestie est, eget tincidunt magna libero et dolor. Fusce tempor sapien et urna sodales, at laoreet nunc egestas. Aenean eget arcu in felis ultrices vehicula. Maecenas interdum viverra sem vitae sodales. Duis tincidunt varius nunc sed viverra. Duis sit amet semper ex, ultrices aliquet mi. Donec a congue sem, ac facilisis ex.',
  'Proin venenatis tortor et ex viverra, ac facilisis mi ullamcorper. Aenean eget est bibendum, fermentum enim eget, semper orci. Nunc magna mauris, fermentum sit amet diam ut, semper malesuada justo. Fusce faucibus tellus eget arcu imperdiet, aliquet placerat massa molestie. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Maecenas gravida quam id ultricies accumsan. Sed porta accumsan libero sed venenatis. Morbi scelerisque, ex in gravida porttitor, ipsum erat sodales urna, mattis mattis nibh ligula quis tellus. Pellentesque consequat et metus scelerisque dapibus. Praesent at maximus ligula. Phasellus vel orci tortor.',
  'Mauris posuere nisl sit amet justo condimentum volutpat. Aliquam venenatis rhoncus est, tristique semper tortor iaculis et. Donec pretium metus nec sodales mollis. Ut vel ipsum sapien. Fusce sit amet libero efficitur orci aliquet scelerisque a ut arcu. In hac habitasse platea dictumst. Sed turpis libero, dictum nec imperdiet nec, maximus eget urna. Sed vel lorem sit amet nisi vulputate mattis. In in odio erat. Aenean euismod lorem nec viverra auctor. Mauris id ipsum maximus, tempus dui nec, mollis felis. Etiam sodales metus sit amet lacus aliquet tristique. Nulla in lacus massa. Morbi quis lorem vitae dolor maximus luctus vel vel eros. Phasellus posuere in purus eget convallis.',
  'Aliquam mauris justo, ornare quis quam fermentum, mollis volutpat dolor. Donec sagittis velit non nisl porttitor, id lacinia tortor vulputate. Suspendisse posuere at felis eu fermentum. Donec at vehicula leo, id sollicitudin ipsum. Donec dapibus sodales erat at malesuada. Mauris sollicitudin velit sed arcu molestie congue. Suspendisse ac arcu enim. Sed ac urna id mi rutrum ornare sit amet in nisl. Phasellus at tempus magna. Nulla facilisi. Mauris pharetra finibus mi ut facilisis. Duis mattis odio id quam vestibulum, id consequat ex fringilla. Aenean sollicitudin accumsan sollicitudin. Duis euismod maximus tempor. Phasellus mi enim, bibendum at nisi vel, cursus venenatis sem. Praesent sagittis, nunc at rhoncus maximus, ligula turpis imperdiet eros, pharetra tincidunt tortor magna id sem.',
  'Sed eleifend sapien condimentum iaculis cursus. In vel pretium turpis. Duis molestie euismod turpis id consectetur. Nunc pharetra sapien libero, a volutpat erat finibus eget. Duis euismod eleifend felis consequat fringilla. Sed varius diam ligula, quis rutrum ex mollis non. Phasellus ac malesuada nibh. Praesent sit amet auctor urna. Quisque eget magna id ex dapibus vulputate eget vel magna.',
  'Ut viverra tempor placerat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras neque diam, auctor eget sapien et, tincidunt blandit turpis. Nam non volutpat velit. Aliquam iaculis euismod nisl non facilisis. Ut blandit dignissim turpis in finibus. Cras neque augue, tempus non neque eu, ornare ultrices sem. Nullam ultricies mi quis semper pellentesque. Mauris mauris lacus, ullamcorper a iaculis quis, tempus id tellus. Phasellus in orci orci. Fusce consequat suscipit ipsum ac bibendum. Maecenas et leo eu leo iaculis varius. Nam consectetur laoreet nisl blandit condimentum.',
  'Ut dictum, sapien vel malesuada posuere, tortor tellus blandit risus, eget hendrerit quam diam eu nibh. Aliquam erat volutpat. Ut rhoncus augue ac nunc porttitor hendrerit. Pellentesque enim risus, auctor quis turpis eget, pulvinar malesuada urna. Nullam tortor mi, elementum id sodales eget, aliquam et nulla. Pellentesque efficitur mi eros, in rutrum metus ultrices vel. Proin vitae diam sit amet nunc pretium malesuada. Fusce at nisi scelerisque, ultrices tortor nec, finibus velit. Donec molestie augue eget leo rutrum, nec ornare arcu gravida. Quisque tincidunt risus elit, ut condimentum eros egestas congue.',
];
export function loremIpsum(paras = 3) {
  let ipsums = [];
  for(let i = 0; i < paras; i++) {
    ipsums.push((
      <p key={i}>{lorems[Math.floor(Math.random() * lorems.length)]}</p>
    ));
  }
  return ipsums;
}

export const numstr = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'];

export const randomImage = ({ width, height, time } = {}) => {
  width = !width ? 200 : width;
  height = !height ? 200 : height;
  time = typeof time === 'undefined' ? Date.now() : time;
  return `https://unsplash.it/${width}/${height}/?random&time=${time}`;
};

export const hostPrefix = ''; // change to '/react-md' for production
export const imgPrefix = hostPrefix + '/imgs';
