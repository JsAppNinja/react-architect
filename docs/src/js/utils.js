import React from 'react';

import { FontIcon, Avatar } from 'react-md';

export const githubHref = 'https://github.com/mlaursen/react-md';

export const mainLinks = [{
  link: '',
  primaryText: 'Home',
  leftIcon: <FontIcon>home</FontIcon>,
}, {
  link: 'sassdoc',
  primaryText: 'SASS Doc',
  leftAvatar: <Avatar src="/imgs/sass-icon.png" alt="SASS Icon" />,
}];
