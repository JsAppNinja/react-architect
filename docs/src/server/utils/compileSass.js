const path = require('path');
const Promise = require('bluebird');
const nodeSass = require('node-sass');

const sass = Promise.promisify(nodeSass.render);

const { ACCENTABLE_COLORS, PRIMARY_COLORS, SECONDARY_HUES } = require('../../shared/constants/colors');

const DARK_THEME_STYLES = `
.example-code,
.code-block {
  background: $md-grey-300;
}

a:not(.md-btn) {
  color: $md-purple-a-400;

  &:visited {
    color: $md-purple-a-100;
  }
}
`;

function toColorName(color) {
  return color && color.replace('_', '-');
}

function validatePrimary(color) {
  return color && PRIMARY_COLORS.indexOf(color) !== -1;
}

function validateSecondary(color, hue) {
  return color && hue && ACCENTABLE_COLORS.indexOf(color) !== -1 && SECONDARY_HUES.indexOf(hue) !== -1;
}

/**
 * Attempts to compiles sass based off of a css file name. The filename should be formatted
 * as 'PRIMARY_COLOR-SECONDARY_COLOR-HUE-dark?.css' where `dark` is optional.
 *
 * @param {String} name - The CSS filename to convert into styles.
 * @return {Promise} a promise with the buffer of compiles styles
 */
module.exports = function compileSass(name) {
  const [p, s, hue, dark] = name.split('-');
  const primary = toColorName(p);
  const secondary = toColorName(s);

  if (!validatePrimary(primary)) {
    return Promise.reject(new Error(`Invalid primary color: ${primary}`));
  } else if (!validateSecondary(secondary, parseInt(hue, 10))) {
    return Promise.reject(new Error(`Invalid secondary color: ${secondary}-${hue}`));
  }

  const primaryColor = `$md-${primary}-500`;
  const secondaryColor = `$md-${secondary}-a-${hue}`;
  const darkThemeStyles = dark ? DARK_THEME_STYLES : '';

  const styles = `
@import 'react-md';

$md-primary-color: $md-light-blue-500;
$md-secondary-color: $md-deep-orange-a-200;
@include react-md-theme-everything(${primaryColor}, ${secondaryColor}, ${!dark}, 'custom-theme');

.custom-theme {
  .banner {
    background: ${primaryColor};
  }

  .react-md-logo {
    text,
    #Group > path:first-child,
    path#Oval-210 {
      fill: ${secondaryColor};
    }
  }

  ${darkThemeStyles}
}

@media #{$md-tablet-media} {
  .custom-theme .phone-size-container .md-toolbar:not(.md-toolbar--inset)::before {
    background: ${primaryColor};
  }
}
  `;

  return sass({
    data: styles,
    includePaths: [path.resolve(process.cwd(), '..', 'src', 'scss')],
    outputStyle: 'compressed',
  }).then((stats, error) => {
    if (error) {
      return Promise.reject(error);
    }

    return stats.css;
  });
};
