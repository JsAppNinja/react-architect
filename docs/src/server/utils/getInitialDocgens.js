import fetch from 'actions/fetch';
import { API_URL } from 'constants/application';

export default async function getInitialDocgens(pathname, component, section) {
  if (!pathname.match(/components/)) {
    return {};
  }

  const ids = [section, component].filter(id => !!id);

  const docgens = await fetch(`${API_URL}/docgens/${ids.join('/')}`);

  if (ids.length === 2) {
    return {
      [ids[0]]: { [ids[1]]: docgens },
    };
  }

  return {
    [ids[0]]: docgens,
  };
}
