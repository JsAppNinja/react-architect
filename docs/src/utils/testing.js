/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { StaticRouter } from 'react-router';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { render, shallow, mount } from 'enzyme';
import rootReducer from 'state';

/**
 * I'm lazy.
 *
 * Could be lazier and do `expectSnapshot` though..
 */
export function createSnapshot(children) {
  return renderer.create(children).toJSON();
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * uses something from react-router.
 */
export function createRouterSnapshot(children, location = '/', context = {}) {
  return createSnapshot(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * requires the redux provider.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function createReduxSnapshot(children, state) {
  const store = createStore(rootReducer, state);
  return createSnapshot(
    <Provider store={store}>
      {children}
    </Provider>
  );
}

/**
 * A simple test wrapper to create a snapshot with react-test-renderer and a component that
 * requires both the redux Provider and something from react-router.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function createReduxRouterSnapshot(children, state, location = '/', context = {}) {
  const store = createStore(rootReducer, state);
  return createSnapshot(
    <Provider store={store}>
      <StaticRouter location={location} context={context}>
        {children}
      </StaticRouter>
    </Provider>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component relies on react-router. This should be used when a component
 * or child component uses findDOMNode or the CSSTransitionGroup, otherwise the createRouterSnapshot should
 * be used.
 */
export function renderRouterSnapshot(children, location = '/', context = {}) {
  return render(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component requires the redux Provider. This should be used when a component
 * or child component uses findDOMNode or the CSSTransitionGroup, otherwise the createReduxSnapshot should
 * be used.
 */
export function renderReduxSnapshot(children, state) {
  const store = createStore(rootReducer, state);
  return render(
    <Provider store={store}>
      {children}
    </Provider>
  );
}

/**
 * This is a simple test wrapper to create a snapshot with enzyme's render instead of react-test-renderer
 * and when the component or child component requires both the redux Provider and something from react-router.
 * This should be used when a component * or child component uses findDOMNode or the CSSTransitionGroup,
 * otherwise the createReduxRouterSnapshot should be used.
 */
export function renderReduxRouterSnapshot(children, state, location = '/', context = {}) {
  const store = createStore(rootReducer, state);
  return render(
    <Provider store={store}>
      <StaticRouter location={location} context={context}>
        {children}
      </StaticRouter>
    </Provider>
  );
}

/**
 * A simple wrapper to use enzyme's shallow rendering when the component or a child
 * requires something from react-router.
 */
export function shallowWithRouter(children, location = '/', context = {}) {
  return shallow(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}


/**
 * A simple wrapper to use enzyme's mount rendering when the component or a child
 * requires something from react-router.
 */
export function mountWithRouter(children, location = '/', context = {}) {
  return mount(
    <StaticRouter location={location} context={context}>
      {children}
    </StaticRouter>
  );
}

/**
 * A simple wrapper to use enzyme's shallow rendering when the component or a child
 * requires something from react-redux.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function shallowWithProvider(children, state) {
  const store = createStore(rootReducer, state);
  return shallow(<Provider store={store}>{children}</Provider>);
}

/**
 * A simple wrapper to use enzyme's mount rendering when the component or a child
 * requires something from react-redux.
 *
 * > It is better to test "pure" components instead of these, so this probably shouldn't
 * > get used much.
 */
export function mountWithProvider(children, state) {
  const store = createStore(rootReducer, state);
  return mount(<Provider store={store}>{children}</Provider>);
}

/**
 * A utility function that will override the console's behavior and throw errors
 * if console.error occurs. This is useful when you want to consider React warnings
 * as failures.
 */
export function captureConsole(match = null) {
  const console = global.console;

  beforeAll(() => {
    global.console = {
      ...console,
      error: (error) => {
        if (match === null || error.match(match)) {
          throw new Error(error);
        }
      },
    };
  });

  afterAll(() => {
    global.console = console;
  });
}
