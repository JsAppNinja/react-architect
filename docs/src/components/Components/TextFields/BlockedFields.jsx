import React, { PureComponent } from 'react';
import {
  Avatar,
  Autocomplete,
  Button,
  Chip,
  Divider,
  TextField,
} from 'react-md';
import guid from 'uuid/v1';

import { randomImage } from 'utils/random';
import PhoneEmulator from 'components/PhoneEmulator';

import './_blocked-styles.scss';

const sendMail = <Button icon>send</Button>;
const contacts = [
  'Albert Brady',
  'Barbara Butler',
  'Benny Warren',
  'Brad Briggs',
  'Brent Allen',
  'Carrie Tyler',
  'Christian Adams',
  'Darrel Mckenzie',
  'Edward Castro',
  'Elsa Mack',
  'Gregg Pearson',
  'Janet Goodwin',
  'Jeanne Hayes',
  'Juan Mccarthy',
  'Julia Robertson',
  'Lola Stephens',
  'Lori Jones',
  'Mable Santiago',
  'Marcia Mills',
  'Margaret Stevenson',
  'Maxine Wilkerson',
  'Randal Newton',
  'Roxanne Ryan',
  'Sergio Hansen',
  'Willie Dawson',
].map(contact => ({
  leftAvatar: <Avatar src={randomImage()} alt={`${contact}'s avatar`} />,
  primaryText: contact,
}));

const defaultBody = `Hi,
I just wanted to check in and see if you had any plans this upcoming weekend. ` +
  'We are thinking of heading up to Napa. Let us know if you\'d like to go and ' +
  'we\'ll make reservations.';

export default class BlockedFields extends PureComponent {
  state = {
    selected: [{
      key: 'initial-1',
      label: 'Freddy Kruger',
      avatar: <Avatar src={randomImage()} alt="Freddy Kruger's avatar" />,
    }, {
      key: 'initial-2',
      label: 'Bob Belcher',
      avatar: <Avatar src={randomImage()} alt="Bob Belcher's avatar" />,
    }],
  };

  selectContact = (name, i, matches) => {
    const contact = matches[i];
    const selected = this.state.selected.slice();
    selected.push({ label: name, avatar: contact.leftAvatar, key: guid() });
    this.setState({ selected });
  };

  removeContact = (i) => {
    const selected = this.state.selected.slice();
    selected.splice(i, 1);
    this.setState({ selected });
  };

  render() {
    const { selected } = this.state;
    let selectedContacts = null;
    if (selected.length) {
      selectedContacts = (
        <div className="text-fields__contacts">
          {selected.map((contact, i) => (
            <Chip
              {...contact}
              className="text-fields__contact-chip"
              removable
              onClick={() => this.removeContact(i)}
            />
          ))}
        </div>
      );
    }

    return (
      <PhoneEmulator transitionContent={false} toolbarActions={sendMail}>
        {selectedContacts}
        <Autocomplete
          id="email-to"
          type="email"
          data={contacts}
          placeholder="To"
          block
          paddedBlock
          onAutocomplete={this.selectContact}
          clearOnAutocomplete
        />
        <Divider />
        <TextField
          id="email-subject"
          placeholder="Subject"
          block
          paddedBlock
          maxLength={80}
          defaultValue="Plans for the weekend"
        />
        <Divider />
        <TextField
          id="email-body"
          placeholder="Body"
          block
          rows={4}
          paddedBlock
          maxLength={1000}
          defaultValue={defaultBody}
          errorText="Max 1000 characters."
        />
      </PhoneEmulator>
    );
  }
}
