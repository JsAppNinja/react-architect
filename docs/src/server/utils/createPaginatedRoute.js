import { BAD_REQUEST } from 'constants/responseCodes';
import { getUrl } from 'server/utils/getServerUrl';
import { omit } from 'lodash';

function isInt(x) {
  return !Number.isNaN(x) && x % 1 === 0;
}

/**
 * Creates an express route handler for paginating data. The paginated data response will return
 * an object containing meta information and the list of data.
 *
 * The metadata will contain the requested start index, the requested limit, the total number of results,
 * an optional next link, and an option previous link. This will allow the client to automatically request
 * the next dataset when needed.
 *
 * @param {function} getData - a function to get the data for the current route. It will provide the current
 *    query string.
 * @param {Object=} config - Any additional config to use.
 * @param {boolean=true} config.queryable - boolean if the route should be queryable.
 * @param {number=50} config.maxResults - the maximun number of results allowed.
 * @param {Array.<String>|String=} config.omitKeys - An optional list or sintr string of keys to remove
 *    from the response.
 * @return {function} a route handler for getting paginated data.
 */
export default function createPaginatedRoute(getData, { queryable = true, maxResults = 50, omitKeys = null } = {}) {
  return function handlePaginationRequest(req, res) {
    const { q } = req.query;
    let { start, limit } = req.query;

    if (!start) {
      start = 0;
    }

    if (!limit) {
      limit = 10;
    }

    if ((queryable && !q) || !isInt(start) || !isInt(limit) || start < 0 || limit > maxResults || limit < 1) {
      res.sendStatus(BAD_REQUEST);
      return;
    }

    start = parseInt(start, 10);
    limit = parseInt(limit, 10);

    const data = getData(q);
    const total = data.length;
    if (start > total) {
      res.sendStatus(BAD_REQUEST);
      return;
    }

    let url = `${getUrl(req).replace(/\?.*/, '')}?`;
    if (queryable) {
      url = `${url}q=${q}&`;
    }

    let next = null;
    let previous = null;
    if (total > start + limit) {
      next = `${url}start=${start + limit}&limit=${limit}`;
    }

    if (start + limit > limit) {
      previous = `${url}start=${Math.max(0, start - limit)}&limit=${limit}`;
    }

    const sliced = data.slice(start, start + limit);

    res.json({
      meta: {
        start,
        limit,
        total,
        next,
        previous,
      },
      data: omitKeys && omitKeys.length ? sliced.map(datum => omit(datum, omitKeys)) : sliced,
    });
  };
}
