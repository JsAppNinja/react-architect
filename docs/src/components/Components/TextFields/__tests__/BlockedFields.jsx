/* eslint-env jest */
import React from 'react';
import { renderReduxSnapshot } from 'utils/testing';

import BlockedFields from '../BlockedFields';

jest.mock('react-md/lib/Helpers/Layover');

describe('BlockedFields', () => {
  it('should render correctly', () => {
    const tree = renderReduxSnapshot(<BlockedFields />);
    expect(tree).toMatchSnapshot();
  });
});
