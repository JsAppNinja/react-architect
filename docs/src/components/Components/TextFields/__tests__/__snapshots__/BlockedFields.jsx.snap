// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockedFields should render correctly 1`] = `
<div
  className="md-paper md-paper--1 phone-emulator md-background"
  style={undefined}
>
  <header
    className="phone-emulator__status-bar"
  >
    <i
      className="md-icon material-icons"
      style={undefined}
    >
      network_wifi
    </i>
    <i
      className="md-icon material-icons"
      style={undefined}
    >
      network_cell
    </i>
    <i
      className="md-icon material-icons"
      style={undefined}
    >
      battery_full
    </i>
    <span
      className="phone-emulator__status-bar__time"
    >
      12:30
    </span>
  </header>
  <header
    className="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--fixed phone-emulator__toolbar"
    style={undefined}
  >
    <button
      className="md-btn md-btn--icon md-pointer--hover md-inline-block md-btn--toolbar md-toolbar--action-left"
      disabled={undefined}
      href={undefined}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <i
        className="md-icon material-icons md-text--inherit"
        style={undefined}
      >
        menu
      </i>
    </button>
    <h2
      className="md-title md-title--toolbar"
      id={undefined}
      style={undefined}
    >
      Title
    </h2>
    <div
      className="md-cell--right md-toolbar--action-right"
    >
      <button
        className="md-btn md-btn--icon md-pointer--hover md-inline-block md-btn--toolbar"
        disabled={undefined}
        href={undefined}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        type="button"
      >
        <i
          className="md-icon material-icons md-text--inherit"
          style={undefined}
        >
          send
        </i>
      </button>
    </div>
  </header>
  <section
    className="phone-emulator__content phone-emulator__content--padded phone-emulator__content--offset"
    id={undefined}
    style={undefined}
  >
    <div
      className="text-fields__contacts"
    >
      <button
        className="md-chip md-chip--avatar md-chip--remove text-fields__contact-chip"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="button"
      >
        <div
          className="md-inline-block md-avatar md-avatar--default"
        >
          <img
            alt="Freddy Kruger's avatar"
            className="md-avatar-img"
            role={undefined}
            src="https://unplash.it/300?random&time=3"
          />
        </div>
        <span
          className="md-chip-text"
        >
          Freddy Kruger
        </span>
        <i
          className="md-icon material-icons md-chip-icon md-chip-icon--rotate"
          style={undefined}
        >
          add_circle
        </i>
      </button>
      <button
        className="md-chip md-chip--avatar md-chip--remove text-fields__contact-chip"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="button"
      >
        <div
          className="md-inline-block md-avatar md-avatar--default"
        >
          <img
            alt="Bob Belcher's avatar"
            className="md-avatar-img"
            role={undefined}
            src="https://unplash.it/300?random&time=3"
          />
        </div>
        <span
          className="md-chip-text"
        >
          Bob Belcher
        </span>
        <i
          className="md-icon material-icons md-chip-icon md-chip-icon--rotate"
          style={undefined}
        >
          add_circle
        </i>
      </button>
    </div>
    <hr
      className="md-divider"
    />
    <div
      className="md-text-field-container md-full-width md-text-field-container--input md-text-field-container--input-block md-text-field-container--padded-block"
      onClick={[Function]}
      onDoubleClick={undefined}
      onMouseDown={undefined}
      onMouseLeave={undefined}
      onMouseOver={undefined}
      onMouseUp={undefined}
      onTouchCancel={undefined}
      onTouchEnd={undefined}
      onTouchMove={undefined}
      onTouchStart={undefined}
      style={
        Object {
          "width": null,
        }
      }
    >
      <input
        className="md-text-field md-full-width md-text"
        defaultValue="Plans for the weekend"
        disabled={undefined}
        id="email-subject"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Subject"
        rows={undefined}
        style={undefined}
        type="text"
      />
      <div
        className="md-text-field-message-container md-text-field-message-container--count-only md-text--disabled"
      >
        <div
          aria-hidden={true}
          className="md-text-field-message md-text-field-message--inactive md-text-field-message--counter"
        >
          21 / 80
        </div>
      </div>
    </div>
    <hr
      className="md-divider"
    />
    <div
      className="md-text-field-container md-full-width md-text-field-container--multiline md-text-field-container--multiline-block md-text-field-container--padded-block"
      onClick={[Function]}
      onDoubleClick={undefined}
      onMouseDown={undefined}
      onMouseLeave={undefined}
      onMouseOver={undefined}
      onMouseUp={undefined}
      onTouchCancel={undefined}
      onTouchEnd={undefined}
      onTouchMove={undefined}
      onTouchStart={undefined}
      style={
        Object {
          "width": null,
        }
      }
    >
      <div
        className="md-text-field-multiline-container"
        style={
          Object {
            "height": null,
          }
        }
      >
        <span
          aria-hidden={true}
        />
        <textarea
          aria-hidden={true}
          className="md-text-field md-text-field--multiline md-full-width md-text md-text-field--multiline-mask"
          defaultValue="Hi,
        I just wanted to check in and see if you had any plans this upcoming weekend. We are thinking of heading up to Napa. Let us know if you'd like to go and we'll make reservations."
          readOnly={true}
          rows={4}
          style={undefined}
          tabIndex={-1}
          value={undefined}
        />
        <textarea
          className="md-text-field md-text-field--multiline md-full-width md-text"
          defaultValue="Hi,
        I just wanted to check in and see if you had any plans this upcoming weekend. We are thinking of heading up to Napa. Let us know if you'd like to go and we'll make reservations."
          disabled={undefined}
          id="email-body"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Body"
          rows={4}
          style={
            Object {
              "height": null,
            }
          }
          value={undefined}
        />
      </div>
      <div
        className="md-text-field-message-container md-text-field-message-container--count-only md-text--disabled"
      >
        <div
          aria-hidden={true}
          className="md-text-field-message md-text-field-message--inactive"
        >
          Max 1000 characters.
        </div>
        <div
          aria-hidden={true}
          className="md-text-field-message md-text-field-message--inactive md-text-field-message--counter"
        >
          181 / 1000
        </div>
      </div>
    </div>
  </section>
</div>
`;
